jar/jar.go

```diff
@@ -249,11 +249,18 @@ func (c *checker) bad() bool {
 
 const bufSize = 4 << 10 // 4 KiB
 
-var bufPool = sync.Pool{
-	New: func() interface{} {
-		return make([]byte, bufSize)
-	},
-}
+var (
+	bufPool = sync.Pool{
+		New: func() interface{} {
+			return make([]byte, bufSize)
+		},
+	}
+	dynBufPool = sync.Pool{
```

Comment:
I didn't reuse it because that bufpool only ever has buffers of `bufSize` (static size). While the "dynamic" buffers are growing. I'm not 100% sure whether two separate pools or a single one would be preferred. I think it might lead to unnecessary memory usage but I'm not certain. Any thoughts @mknyszek?

Reply:
Ah, I thought you were just using dynBufPool for an optimization when handling small buffers. Not storing dynamically sized []byte slices in the pool.

I think storing dynamic buffers in sync.Pool is arguably incorrect? See https://github.com/golang/go/issues/23199

Please be aware that this method may be called from long-lived server contexts, not just command line tools, so we have to be cautions.

Reply:
OK, in order to mitigate this for long-lived servers, I've added a sync.Pool wrapper that should remedy this. PTAL. I've also added relevant benchmarks to the new commit message.